debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresSnapshotChangeEventSource.java

```diff
@@ -242,20 +244,34 @@ protected void aborted(SnapshotContext<PostgresPartition, PostgresOffsetContext>
     @Override
     protected Optional<String> getSnapshotSelect(RelationalSnapshotContext<PostgresPartition, PostgresOffsetContext> snapshotContext,
                                                  TableId tableId, List<String> columns) {
-        if (snapshotContext.onDemand) {
-            return blockingSnapshotter.buildSnapshotQuery(tableId, columns);
-        }
 
-        return snapshotter.buildSnapshotQuery(tableId, columns);
+        return snapshotterService.getSnapshotQuery().snapshotQuery(tableId.toDoubleQuotedString(), columns);
     }
 
     protected void setSnapshotTransactionIsolationLevel(boolean isOnDemand) throws SQLException {
         LOGGER.info("Setting isolation level");
-        String transactionStatement = snapshotter.snapshotTransactionIsolationLevelStatement(slotCreatedInfo, isOnDemand);
+        String transactionStatement = snapshotTransactionIsolationLevelStatement(slotCreatedInfo, isOnDemand);
         LOGGER.info("Opening transaction with statement {}", transactionStatement);
         jdbcConnection.executeWithoutCommitting(transactionStatement);
     }
 
+    private String snapshotTransactionIsolationLevelStatement(SlotCreationResult newSlotInfo, boolean isOnDemand) {
+
+        if (newSlotInfo != null && !isOnDemand) {
+            /*
+             * For an on demand blocking snapshot we don't need to reuse
+             * the same snapshot from the existing exported transaction as for the initial snapshot.
+             */
+            String snapSet = String.format("SET TRANSACTION SNAPSHOT '%s';", newSlotInfo.snapshotName());
+            return "SET TRANSACTION ISOLATION LEVEL REPEATABLE READ; \n" + snapSet;
+        }
+
+        // TODO should this customizable?
+
+        // we're using the same isolation level that pg_backup uses
+        return "SET TRANSACTION ISOLATION LEVEL SERIALIZABLE, READ ONLY, DEFERRABLE;";
```

Comment:
@jpechane Should we let customize it? This seems to be valid only for PostgresSQL and MySQL.

Reply:
No, let's keep it per connector now. We can always reconsider in future